udata/core/dataset/models.py

```diff
@@ -104,6 +110,28 @@ class License(db.Document):
     def __unicode__(self):
         return self.title
 
+    @classmethod
+    def guess(cls, text):
+        '''
+        Try to guess license from a string.
+
+        Try to exact match on identifier then slugified title
+        and fallback on edit distance ranking (after slugification)
+        '''
+        qs = cls.objects
+        text = text.strip().lower()  # Stored identifiers are lower case
+        slug = cls.slug.slugify(text)  # Use slug as it normalize string
+        license = qs(db.Q(id=text) | db.Q(slug=slug) | db.Q(url=text)).first()
+        if license is None:
+            # Try to single match with a low Damerau-Levenshtein distance
+            computed = ((l, rdlevenshtein(l.slug, slug)) for l in cls.objects)
+            candidates = [l for l, d in computed if d <= MAX_DISTANCE]
+            # If there is more that one match, we cannot determinate
+            # which one is closer to safely chose between candidates
```

Comment:
No, we don't and let the license unknown.
We don't chose a license for the producer because if we chose something wrong (ie. a license that is more permissive that the original one), we are legaly responsible.
Example: there is some bug differences between cc-by-sa 2.0, 3.0 et 4.0 but if someone type cc-by-sa 5.0 (which doesn't exists), I don't want to choose arbitrary between the different versions.

This is a first implementation but we already discussed some possible improvements for some other existing case:
- store a list of known synonyms for each license (ie. License Etalab for License ouverte)
- have some rule for license version handling (ie. CC-BY-SA without version is latest 4.0 version)